app/Http/Controllers/Api/v1/ReminderApiController.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace App\Http\Controllers\Api\v1;
+
+use App\Http\Controllers\Api\v1\ApiController;
+use App\Http\Requests\API\StoreReminder as Request;
+use App\Http\Resources\ReminderResource as Resource;
+use App\Models\Campaign;
+use App\Models\Entity;
+use App\Models\Reminder;
+
+class ReminderApiController extends ApiController
+{
+    /**
+     * @return \Illuminate\Http\Resources\Json\AnonymousResourceCollection
+     *
+     * @throws \Illuminate\Auth\Access\AuthorizationException
+     */
+    public function index(Campaign $campaign, Entity $entity)
+    {
+        $this->authorize('access', $campaign);
+        $this->authorize('view', $entity);
+
+        return Resource::collection($entity->reminders()->paginate());
+    }
+
+    /**
+     * @return resource
+     */
+    public function show(Campaign $campaign, Entity $entity, Reminder $reminder)
+    {
+        $this->authorize('access', $campaign);
+        $this->authorize('view', $entity);
+
+        return new Resource($reminder);
+    }
+
+    /**
+     * @return resource
+     *
+     * @throws \Illuminate\Auth\Access\AuthorizationException
+     */
+    public function store(Request $request, Campaign $campaign, Entity $entity)
+    {
+        $this->authorize('access', $campaign);
+        $this->authorize('update', $entity);
+        $data = $request->all();
+        $data['entity_id'] = $entity->id;
+        $model = Reminder::create($data);
+        $model->refresh();
+
+        return new Resource($model);
+    }
+
+    /**
+     * @return resource
+     */
+    public function update(Request $request, Campaign $campaign, Entity $entity, Reminder $reminder)
+    {
+        $this->authorize('access', $campaign);
```

Comment:
Missing check that the reminder belongs to the entity. I can now delete a reminder of a post/other entity I don't have access to.